src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -86,18 +86,21 @@ internal abstract class PENamedTypeSymbol : NamedTypeSymbol
         private static readonly UncommonProperties s_noUncommonProperties = new UncommonProperties();
         private UncommonProperties _lazyUncommonProperties;
 
-        private UncommonProperties GetUncommonProperties()
+        private UncommonProperties GetUncommonProperties(bool force = false)
         {
             var result = _lazyUncommonProperties;
             if (result != null)
             {
 #if DEBUG
+                // We require that if 'force: true' is passed, then we must not have already loaded the default instance.
+                // We want to avoid a scenario where calling multiple times with 'force: false' could cause the caller to observe different instances.
+                Debug.Assert(!force || result != s_noUncommonProperties);
```

Comment:
>Debug.Assert(!force || result != s_noUncommonProperties); [](http://example.com/codeflow?start=16&length=57)

I do not like this approach. An assert is not good enough, in my opinion. If our assumption is to use ```force == true``` only once before any one else calls this helper. Then, I would prefer not modifying this helper and not using it there. Instead, I would unconditionally throw if ```_lazyUncommonProperties``` is not null there. Unconditionally meaning, perform the check regardless whether we want to crate ```UncommonProperties```. And if we determine that we do want to create them, we should simply assign directly to ```_lazyUncommonProperties```. Since this is happening in context of a constructor, we don't need to worry about threading issues.